modules/vaos/app/models/vaos/appointment_request_form.rb

```diff
@@ -0,0 +1,155 @@
+    attribute :purpose_of_visit, String
+    attribute :provider_id, String
+    attribute :second_request, Boolean
+    attribute :second_request_submitted, Boolean
+    attribute :provider_name, String
+    attribute :requested_phone_call, Boolean
+    attribute :type_of_care_id, Boolean
+    attribute :has_veteran_new_message, Boolean
+    attribute :has_provider_new_message, Boolean
+    attribute :provider_seen_appointment_request, Boolean
+    attribute :requested_phone_call, Boolean
+    attribute :type_of_care_id, String
+    attribute :patient_id, String # This setter is overriden
+    attribute :appointment_request_id, String # This setter is overriden
+    attribute :unique_id, String # This setter is overriden
+    attribute :id, String # This setter is overriden
+    attribute :date, String # Automatically set for create if id is present; This setter is overridden
+    attribute :created_date, VAOS::AppointmentTime # Only passed in when updating
+    attribute :last_access_date, VAOS::AppointmentTime # Only passed in when updating
+    attribute :last_updated_date, VAOS::AppointmentTime # Only passed in when updating
+    attribute :assigning_authority, String, default: 'ICN'
+    attribute :system_id, String, default: 'var'
+    attribute :object_type, String, default: 'VARAppointmentRequest'
+    attribute :surrogate_identifier, Hash, default: {}
+    attribute :facility, Hash
+    attribute :patient, Hash, default: { inpatient: false, text_messaging_allowed: false } # This setter is overriden
+    attribute :best_timeto_call, Array[String] # VAR camel cases this incorrectly so we will snakecase it incorrectly
+    attribute :appointment_request_detail_code, Array[String] # This setter is overriden
+
+    def initialize(user, json_hash)
+      @user = user
+      @id = json_hash[:id]
+      @appointment_request_id = @id # ensure these are the same (and the FE doesn't have to pass in redundant data)
+      @unique_id = @id # ensure these are the same (and the FE doesn't have to pass in redundant data)
+      @date = Time.current.strftime('%Y-%m-%dT%H:%M:%S.%L%z') if @id.blank? # only set for create in proper format
+      @patient_id = edipi # TODO: discover if the patient_id ever comes from elsewhere or if it is always EDIPI
+      super(json_hash)
+    end
+
+    # Ensure that these cannot be mass assigned or modified after initialization
+    def id=(_value); end
+
+    def appointment_request_id=(_value); end
+
+    def unique_id=(_value); end
+
+    def date=(_value); end
+
+    def patient_id=(_value); end
+
+    # These values are provided directly from FE without coercion; we may need to validate some of these
+    def facility=(values_hash)
+      @facility = values_hash.empty? ? {} : values_hash.merge(object_type: 'Facility').compact
+    end
+
+    # These values ought to be derived from MVI vs user input, except for inpatient and text_messaging_allowed
+    def patient=(values_hash)
+      @patient = {
```

Comment:
Feels like this should be a model/struct? A `Patient` model? How can we test behavior on this?

Reply:
It was originally. There was also a Facility model. And I'll probably go back to that once it comes time to add validations as it will make it easier. But for keeping this PR light / small I went with this approach.